packages/flutter/lib/src/material/bottom_app_bar.dart

```diff
@@ -0,0 +1,135 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/rendering.dart';
+import 'package:flutter/widgets.dart';
+
+import 'material.dart';
+import 'scaffold.dart';
+
+// Examples can assume:
+// Widget bottomAppBarContents;
+
+/// A container that shows in the [Scaffold.bottomNavigationBar] slot, and can
+/// have a notch in which a [FloatingActionButton] is showing.
+///
+/// Typically used with a [Scaffold] and a [FloatingActionButton].
+///
+/// ## Sample code
+///
+/// ```dart
+/// new Scaffold(
+///   bottomNavigationBar: new BottomAppBar(
+///     color: Colors.white,
+///     child: bottomAppBarContents,
+///   ),
+///   floatingActionButton: new FloatingActionButton(onPressed: null),
+/// )
+/// ```
+///
+/// See also:
+///
+///   * [NotchMaker] a closure used for creating a notch in a shape.
+///   * [ScaffoldGeometry.floatingActionBarNotchMaker] the [NotchMaker] used to
+///   make a notch for the [FloatingActionButton]
+///   * [FloatingActionButton] which the [BottomAppBar] makes a notch for.
+class BottomAppBar extends StatefulWidget {
+  const BottomAppBar({
```

Comment:
missing api docs; please check the output of `flutter analyze --flutter-repo --dartdocs` and use the prompts at https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#avoid-useless-documentation for ideas of what to write.

Reply:
done